tests/vmemcache_fragmentation.c

```diff
@@ -0,0 +1,279 @@
+		return 1;
+
+	time(&end);
+
+	diff = (int)difftime(end, start);
+	if (diff >= timeout) {
+		return 0;
+	}
+	return 1;
+}
+
+/* Parse cmd args */
+test_params
+parse_args(int argc, char **argv)
+{
+	test_params p = {.pool_size = VMEMCACHE_MIN_POOL,
+		.segment_size = 16,
+		.val_max = 0,
+		.dir = "/tmp",
+		.csv = "",
+		.seconds = 0,
+		.minutes = 0,
+		.hours = 0
+	};
+	size_t val_max_factor = 70;
+
+	static const char *usage_str = "usage: %s "
+	    "[-h] "
+	    "[-p <pool_size>] "
+	    "[-s <segment_size>] "
+	    "[-v <val_max_factor>] "
+	    "[-t <timeout_seconds>] "
+	    "[-m <timeout_minutes>] "
+	    "[-o <timeout_hours>] " "[-c <csv>] " "[-d <dir>]\n ";
+
+	const char *optstr = "hp:s:v:t:m:o:d:c:";
+	int opt;
+	while ((opt = getopt(argc, argv, optstr)) != -1) {
+		char *endptr;
+		switch (opt) {
+		case 'h':
+			printf(usage_str, argv[0]);
+			exit(0);
+		case 'p':
+			p.pool_size = (size_t) (strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid pool size provided");
+			break;
+		case 's':
+			p.segment_size =
+			    (size_t) (strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid segment size provided");
+			break;
+		case 'v':
+			val_max_factor =
+			    (size_t) (strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid val max factor ovided");
+			break;
+		case 't':
+			p.seconds = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid seconds provided");
+			break;
+		case 'm':
+			p.minutes = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid minutes provided");
+			break;
+		case 'o':
+			p.hours = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid hours provided");
+			break;
+		case 'd':
+			strcpy(p.dir, optarg);
+			break;
+		case 'c':
+			strcpy(p.csv, optarg);
+			break;
+		}
+	}
+	p.val_max = val_max_factor * p.segment_size;
+
+	return p;
+}
+
+int
+main(int argc, char **argv)
+{
+	test_params p = parse_args(argc, argv);
+
+	char *val = NULL;
+	FILE *fcsv = NULL;
+	VMEMcache *vc = NULL;
+
+	/* init pool */
+	vc = vmemcache_new(p.dir, p.pool_size, p.segment_size, 1);
+
+	if (!vc)
+		FATAL("vmemcache_new: %s", p.dir);
+
+	on_evict_info info = { 0, 0, false };
+	vmemcache_callback_on_evict(vc, on_evict, &info);
+
+
+	/* create log file */
+	if (strlen(p.csv) != 0) {
+		fcsv = fopen(p.csv, "w");
+		if (!fcsv) {
+			perror(p.csv);
+			goto err;
+		}
+		fprintf(fcsv, "keynum,ratio\n");
+	}
+
+	size_t keynum = 0;
+	size_t keysize = sizeof(size_t);
+
+	float prev;
+	float ratio = 0.0f;
+
+	long seed = time(NULL);
+	srand((unsigned)seed);
+
+	time_t start;
+	time(&start);
+	while (timeout(start, p.seconds, p.minutes, p.hours)) {
+
+		/* create key */
+		char key[keysize];
+		sprintf(key, "%zu", keynum);
+
+		/* generate val */
+		size_t val_size = get_rand_size(p.val_max, p.segment_size);
+
+		val = malloc(val_size);
+		if (!val) {
```

Comment:
You are inconsistent with braces. Single line statements like this one do not need { }.